actiontext/lib/action_text/attribute.rb

```diff
@@ -24,7 +24,7 @@ module Attribute
       #
       #   Message.all.with_rich_text_content # Avoids N+1 queries when you just want the body, not the attachments.
       #   Message.all.with_rich_text_content_and_embeds # Avoids N+1 queries when you just want the body and attachments.
-      def has_rich_text(name)
+      def has_rich_text(name, encrypted: false)
```

Comment:
Would this macro be able to accept `deterministic: true`?

Reply:
For some background, I've opened https://github.com/rails/rails/pull/41627 in an attempt to make the plain text generated by Rich Text records available for full-text searching. Full-text search across that column would probably require the same measures as querying other encrypted columns.

Reply:
Not right now, from [the guide](https://github.com/basecamp/rails/blob/active-record-encryption/guides/source/active_record_encryption.md#action-text):

> NOTE: Passing individual encryption options to action text attributes is not supported yet. It will use non-deterministic encryption with the global encryption options configured.

It would be a very nice thing to add, but it's would also be complex, while the current implementation is very simple. I think a case where you want to do exact matches on a rich text field should be rare, but it's also true that being consistent would be great.

Reply:
> I think a case where you want to do exact matches on a rich text field should be rare

I agree, the HTML debris makes querying that column unreasonable. #41627 introduces a companion column that has the rich text translated via [#to_plain_text](https://edgeapi.rubyonrails.org/classes/ActionText/RichText.html#method-i-to_plain_text). At that point, would encrypting the plain text require the same inputs and configurations as any other `text` column?

Reply:
I think this is something I can experiment with after this ships, but I just want to make sure there aren't any technical or philosophical limitations that would make that sort of thing impossible.

Reply:
How does database-native full-text search against an encrypted column work?

Reply:
> How does database-native full-text search against an encrypted column work?

@georgebrock great question. For something that's _actually_ full-text like Postgres (with ts_vectors and whatnot), it's probably a non-starter and impossible.

While #41627's PR description has a PG example, the changeset itself doesn't tie itself to any database-specific search technique. There's could still value in being able to do a `where("ILIKE ...")` style query against the plain-text column, which seems to be more similar to the `find_by` example shared in this PR.

Reply:
@seanpdoyle `ILIKE` won't work with encrypted content either, unfortunately. 

As mentioned, I agree that having a consistent interface would be great. IMO, more than for the `deterministic:` flag, for all the rest of options we support in `.encrypts`. The problem is that it would be a big change. Action text attributes are a wrapper for an association to a separated record (`ActionText::RichText`). All the rich text attributes points to the same record. This PR adds a new record that encrypts the `body`, and then switches to one or other depending on the `encrypt:` flag. Supporting more fine-grained options would require a different (much more complex) approach.

Reply:
`LIKE` queries won’t work against deterministically-encrypted columns, either, though. The database can’t partially match ciphertext.

Sorry to distract from the original question. I just want to be clear that DB-level full-text search and partial matching won’t work. Data you need to search against or partially match isn’t suitable for (this kind of?) encryption.